# Introdução

Este  trabalho  foi  realizado  no  âmbito  da  unidade  curricular  Análise  Numérica,  com  o 
objetivo  de  apresentar  propostas  de  resolução  de  exercícios  relativos  à  interpolação 
polinomial e à interpolação polinomial segmentada. Estes processos permitem 
aproximar uma função 𝑓, da qual se conhece apenas um conjunto de 𝑛+1 pontos de 
abcissas distintas, a partir de um polinómio interpolador e de um spline, 
respetivamente. Para isso, foram implementados o método de Newton em diferenças 
divididas e a construção do spline cúbico natural, recorrendo à linguagem Python. 

## Exercício 1
#### **a)** Seja F(x) = $x^2 +sen(6x)$, para $−1 ≤ x ≤ 1$

In [4]:
import numpy as np
import math

# Definição da função f(x)
def f(x):
    return x**2 + math.sin(6 * x)

# Parâmetros: intervalo [-1, 1] e 8 pontos equidistantes
a, b = -1, 1
num_pontos = 8
x = np.linspace(a, b, num_pontos)  # Gerar abcissas equidistantes
y = [f(xi) for xi in x]            # Calcular valores da função

# Impressão da tabela com 6 casas decimais
print(" i |     x_i      |     f(x_i)    ")
print("-"*40)
for i in range(8):
    print(f"{i:2d} | {x[i]:+12.6f} | {y[i]:+12.6f}")

 i |     x_i      |     f(x_i)    
----------------------------------------
 0 |    -1.000000 |    +1.279415
 1 |    -0.714286 |    +1.420551
 2 |    -0.428571 |    -0.356097
 3 |    -0.142857 |    -0.735567
 4 |    +0.142857 |    +0.776384
 5 |    +0.428571 |    +0.723444
 6 |    +0.714286 |    -0.400143
 7 |    +1.000000 |    +0.720585



### 1. Contextualização e Objetivo  
O item visa gerar \( n+1 = 8 \) pontos \((x_i, f(x_i))_{i=0}^7\) que definem uma partição equidistante de \([-1, 1]\). Esses pontos são prerequisitos para a construção do polinômio interpolador e spline cúbico natural nos itens 1.(b) e 1.(c), pois ambas as aproximações dependem de pontos conhecidos da função original.  

### 2. Parâmetros Matemáticos Fundamentais  
- **Número de subintervalos**: \( n = 7 \) (derivado de \( n+1 = 8 \) pontos).  
- **Passo equidistante (\( h \))**: Calculado por \( h = \frac{b - a}{n} = \frac{1 - (-1)}{7} = \frac{2}{7} \approx 0.2857 \), garantindo espaçamento uniforme entre abcissas.  
- **Fórmula das abcissas**: \( x_i = a + i \cdot h \) (progressão aritmética), com \( a = -1 \), \( b = 1 \) e \( i = 0, 1, ..., 7 \) (ex.: \( x_0 = -1.0 \), \( x_4 \approx 0.1429 \), \( x_7 = 1.0 \)).  

### 3. Validade da Função \( f(x) \)  
\( f(x) = x^2 + \sin(6x) \) é contínua e diferenciável em \([-1, 1]\) (soma de \( x^2 \) — polinômio, portanto contínuo — e \( \sin(6x) \) — trigonométrica, também contínua). Essa propriedade evita descontinuidades que comprometam a qualidade das aproximações subsequentes.  

### 4. Precisão e Relevância Prática  
- Os valores de \( f(x_i) \) são apresentados com 6 casas decimais, equilibrando clareza e precisão para os cálculos de interpolação.  
- O uso de `numpy.linspace` no código garante geração precisa das abcissas, sem erros de arredondamento associados a cálculos manuais.  

### 5. Ligação com Itens Seguintes  
Os pontos gerados são a entrada para:  
- 1.(b): Construir o polinômio interpolador (grau \( \leq 7 \), pois passa por 8 pontos) e o spline cúbico natural (7 segmentos de grau 3).  
- 1.(c): Calcular majorantes de erros para \( f(0.1) \) e \( f(0.9) \), já que sem \( (x_i, f(x_i)) \) não é possível definir \( p(x) \) nem \( S(x) \).

#### **b)**